client/data/marketplace/use-es-query.ts

```diff
@@ -4,36 +4,55 @@ import { extractSearchInformation } from 'calypso/lib/plugins/utils';
 import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
 import { DEFAULT_PAGE_SIZE } from './constants';
 import { search } from './search-api';
-import { ESHits, ESResponse, Plugin, PluginQueryOptions } from './types';
 import { getPluginsListKey } from './utils';
+import type { ESHits, ESResponse, Plugin, PluginQueryOptions, Icon } from './types';
 
-// /**
-//  *
-//  * @param pluginSlug
-//  * @param iconsArray
-//  * @returns An object containing a resolution -> icon url map
-//  */
-// const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
-// 	type Icon = {
-// 		resolution: string;
-// 		filename: string;
-// 		location: string;
-// 		revision: string;
-// 	};
-// 	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
-// 	return Object.values( icons ).reduce(
-// 		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
-// 			prev[
-// 				resolution
-// 			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
-// 			return prev;
-// 		},
-// 		{}
-// 	);
-// };
+/**
+ *
+ * @param pluginSlug
+ * @param icons
+ * @returns A string containing an icon url or
+ * 	a default generated url Icon if icons param is falsy, it is not JSON or it does not contain a valid resolution
+ */
+const createIconUrl = ( pluginSlug: string, icons?: string ): string => {
+	const defaultIconUrl = buildDefaultIconUrl( pluginSlug );
+	if ( ! icons ) return defaultIconUrl;
 
-// const createAuthorUrl = ( headerAuthor: string, headerAuthorUri: string ) =>
-// 	`<a href="${ headerAuthorUri }">${ headerAuthor }</a>`;
+	let iconsObject: Record< string, Icon > = {};
+	try {
+		iconsObject = JSON.parse( icons );
+	} catch ( error ) {
+		return defaultIconUrl;
+	}
+
+	// Transform Icon response for easier handling
+	const iconByResolution = Object.entries( iconsObject ).reduce( ( icon, iconByResolution ) => {
+		const [ , currentIcon ] = iconByResolution;
+		const newKey = currentIcon.resolution;
+		icon[ newKey ] = currentIcon;
+		return icon;
+	}, {} as Record< string, Icon > );
```

Comment:
As we're reducing into the `iconByResolution` object here, for readability's sake, can we perhaps rename the accumulator/previousValue and item/currentValue params to make it more clear which one is which? 

```suggestion
	// Transform Icon response for easier handling
	const iconByResolution = Object.entries( iconsObject ).reduce(
		( iconByResolution, iconEntry ) => {
			const [ , currentIcon ] = iconEntry;
			const newKey = currentIcon.resolution;
			iconByResolution[ newKey ] = currentIcon;
			return iconByResolution;
		},
		{} as Record< string, Icon >
	);
```

Alternatively, we could also reduce only on the values of the object, to make the function body a bit simpler:

```suggestion
// Transform Icon response for easier handling
const iconMap = Object.values(iconsObject).reduce(
  (iconByResolution, currentIcon) => {
    const newKey = currentIcon.resolution;
    iconByResolution[newKey] = currentIcon;
    return iconByResolution;
  },
  {} as Record<string, Icon>
);
```

Reply:
Absolutely agree, `Object.entries` is not eventually required and the suggestion improves the readability. I have committed it locally to apply the formatting. Thanks for that!